cirq/study/sweeps.py

```diff
@@ -439,3 +435,43 @@ def _params_without_symbols(resolver: resolver.ParamResolver) -> Params:
         if isinstance(sym, sympy.Symbol):
             sym = sym.name
         yield cast(str, sym), cast(float, val)
+
+
+def dict_to_product_sweep(
+        factor_dict: Dict['cirq.TParamKey',
```

Comment:
I think having a named ZipOrProductSweepLike type alias would make this more readable, and signal it more clearly that the two methods are working off of the same type.

Reply:
Done.